### PR TITLE
Adds mouth

### PR DESCRIPTION
**Pull request by UndeadNappist**
Damned whitespacing package.

Supposedly adds the mouth slot.

### CODE REVIEW

**Comment by luciensadi** [comment]
This won't work as written. The new mouth slot must be at the end of all of the arrays-- end of wear_bits, end of the array in constants.cpp, etc. Remember that the code will look values up in the array by the values you provided in awake.h, so it'll return the 40th string in the list for wear location, etc.

**Comment by UndeadNappist** [comment]
Ah, thanks. Figures I was doing something wrong.

File: src/awake.h [hunk 1]
```diff
@@ -1122,6 +1123,7 @@ enum {
 #define ITEM_WEAR_FACE   23
 #define ITEM_WEAR_THIGH  24
 #define ITEM_WEAR_MAX    25
```
Comment (luciensadi) [inline src/awake.h:1125]: X_MAX is used by the code to indicate 'the acceptable values for this are 0 up to this number', so having the mouth flag after wear_max means it won't be accounted for in a lot of the code. Swap these so mouth is 25 and max is 26.

File: src/act.obj.cpp [hunk 2]
```diff
@@ -45,7 +45,7 @@ void calc_weight(struct char_data *ch);
 SPECIAL(weapon_dominator);
 
 int wear_bitvectors[] = {
-                          ITEM_WEAR_TAKE, ITEM_WEAR_HEAD, ITEM_WEAR_EYES, ITEM_WEAR_EAR,
+                          ITEM_WEAR_TAKE, ITEM_WEAR_HEAD, ITEM_WEAR_EYES, ITEM_WEAR_EAR, ITEM_WEAR_MOUTH,
```
Comment (luciensadi) [inline src/act.obj.cpp:48]: This new value needs to be at the end of the wear_bitvectors[] array, after wear_feet. This ensures that we don't mess up any existing wear locations, and also that your string for wearing the item (which you set below) is in the right spot.

File: src/act.obj.cpp [hunk 3]
```diff
@@ -2400,7 +2403,8 @@ void perform_wear(struct char_data * ch, struct obj_data * obj, int where)
                               "YOU SHOULD NEVER SEE THIS MESSAGE.  PLEASE REPORT.\r\n",
                               "You already have something on each of your ankles.\r\n",
                               "You are already wearing something on your feet.\r\n",
-                              "You're already wearing something on your feet.\r\n"
+                              "You're already wearing something on your feet.\r\n", // this message is repeated, not sure why
```
Comment (luciensadi) [inline src/act.obj.cpp:2406]: Socks and shoes.

File: src/awake.h [hunk 4]
```diff
@@ -263,8 +263,9 @@ enum {
 #define WEAR_RANKLE    35
 #define WEAR_SOCK      36
 #define WEAR_FEET      37
-#define WEAR_PATCH     38      
+#define WEAR_PATCH     38
 #define NUM_WEARS      39
```
Comment (luciensadi) [inline src/awake.h:267]: Same comment as below-- NUM_WEARS must be last, so swap this with WEAR_MOUTH.

File: src/act.obj.cpp (hunk 2 above)
Comment (UndeadNappist) [reply src/act.obj.cpp]: I think I did it right this time?

File: src/act.obj.cpp [hunk 5]
```diff
@@ -2400,7 +2404,7 @@ void perform_wear(struct char_data * ch, struct obj_data * obj, int where)
                               "YOU SHOULD NEVER SEE THIS MESSAGE.  PLEASE REPORT.\r\n",
                               "You already have something on each of your ankles.\r\n",
                               "You are already wearing something on your feet.\r\n",
-                              "You're already wearing something on your feet.\r\n"
+                              "You already have something in your mouth.\r\n",
```
Comment (luciensadi) [inline src/act.obj.cpp:2407]: You've replaced the wear-feet message with the mouth one, so people with shoes on who try to wear more shoes will wonder why they're trying to eat their shoes. Recommend restoring `"You're already wearing something on your feet.\r\n",` before this.

Comment (luciensadi) [reply src/act.obj.cpp:2407]: Other than this, it looks good!

Comment (UndeadNappist) [reply src/act.obj.cpp:2407]: Fixed the dangling comma while I was at it.

**Comment by luciensadi** [comment]
Thanks!

**Pull request merged**